src/Common/getNumberOfPhysicalCPUCores.cpp

```diff
@@ -48,7 +48,7 @@ static unsigned getNumberOfPhysicalCPUCoresImpl()
     /// Let's limit ourself to the number of physical cores.
     /// But if the number of logical cores is small - maybe it is a small machine
     /// or very limited cloud instance and it is reasonable to use all the cores.
-    if (cpu_count >= 8)
+    if (cpu_count >= 32)
```

Comment:
What CPUs are you targeting? Probably some AWS instances?

Reply:
I've tested on ec2 instances with 16 logical cores and both amd and intel. and on my vm with 32-core intel proc. in all three cases I saw an improvement when I used the number of logical cores as `max_threads`. so it looks like we are safe to more threshold to the right, at least to 32.